File: web/src/api/aiCoachPreference.ts
import request from "./request";
import type { AICoachPreference, AICoachPreferencePayload } from "@/types/models";

export function getAICoachPreference() {
  return request.get<AICoachPreference>("/ai-coach-preference");
}

export function updateAICoachPreference(payload: AICoachPreferencePayload) {
  return request.put<AICoachPreference>("/ai-coach-preference", payload);
}

File: server/api/routes/ai_coach_preference.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from planner_core.database.models import UserAccount
from server.api.deps import get_current_user, get_db
from server.schemas.ai_coach_preference import AICoachPreferenceRead, AICoachPreferenceUpdate
from server.services import ai_coach_preference_service

router = APIRouter(prefix="/ai-coach-preference", tags=["AI coach preference"])


@router.get("", response_model=AICoachPreferenceRead)
def get_ai_coach_preference(
    db: Session = Depends(get_db),
    current_user: UserAccount = Depends(get_current_user),
):
    return ai_coach_preference_service.get_or_create_preference(db, current_user.id)


@router.put("", response_model=AICoachPreferenceRead)
def update_ai_coach_preference(
    payload: AICoachPreferenceUpdate,
    db: Session = Depends(get_db),
    current_user: UserAccount = Depends(get_current_user),
):
    return ai_coach_preference_service.update_preference(db, current_user.id, payload)
